instrumentation/cassandra/cassandra-4.0/src/main/java/io/opentelemetry/javaagent/instrumentation/cassandra/v4_0/CassandraDatabaseClientTracer.java

```diff
@@ -47,8 +47,10 @@ protected InetSocketAddress peerAddress(CqlSession cqlSession) {
   public void onResponse(Span span, ExecutionInfo executionInfo) {
     Node coordinator = executionInfo.getCoordinator();
     if (coordinator != null) {
-      Optional<InetSocketAddress> address = coordinator.getBroadcastRpcAddress();
-      address.ifPresent(inetSocketAddress -> NetPeerUtils.setNetPeer(span, inetSocketAddress));
+      SocketAddress socketAddress = coordinator.getEndPoint().resolve();
```

Comment:
I don't think it's related to this PR since we have many instrumentations with the same pattern - I often worry about blocking an event loop with a DNS lookup just to set a span attribute. I want (someone) to remove the spec's language encouraging a DNS lookup just to resolve an attribute.

Just the word resolve makes it more obvious than Java where a seemingly normal getter has the chance to block and brought this point up for me :)

Reply:
Good point. But this may be related to the general question of bringing performance considerations into semantic conventions...

Reply:
I'd be interested if we force DNS resolution anywhere, I know we've tried to avoid that in the past.

In this case, I'm not sure why the method is named `resolve()`, at least the implementation I'm seeing when I look doesn't seem to resolve anything.